src/main/java/org/skriptlang/skript/bukkit/recipes/MutableRecipe.java

```diff
@@ -0,0 +1,411 @@
+package org.skriptlang.skript.bukkit.recipes;
+
+import org.bukkit.NamespacedKey;
+import org.bukkit.inventory.*;
+import org.bukkit.inventory.recipe.CookingBookCategory;
+import org.bukkit.inventory.recipe.CraftingBookCategory;
+import org.jetbrains.annotations.Nullable;
+import org.skriptlang.skript.bukkit.recipes.RecipeUtils.RecipeType;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Mutable Recipe class allowing data to be set before creation of a recipe.
+ */
+public abstract class MutableRecipe implements Recipe {
+
+	private final RecipeType recipeType;
+	private final NamespacedKey key;
+	private ItemStack result = null;
+	private final List<String> errors = new ArrayList<>();
+
+	public MutableRecipe(NamespacedKey key, RecipeType recipeType) {
+		this.key = key;
+		this.recipeType = recipeType;
+	}
+
+	public void setResult(ItemStack result) {
+		this.result = result;
+	}
+
+	@Override
+	public @Nullable ItemStack getResult() {
+		return result;
+	}
+
+	public NamespacedKey getKey() {
+		return this.key;
+	}
+
+	public RecipeType getRecipeType() {
+		return recipeType;
+	}
+
+	public void addError(String error) {
+		errors.add(error);
+	}
+
+	public List<String> getErrors() {
```

Comment:
unused. are these errors for an init? otherwise this can be removed

Reply:
They're used within SecRegisterRecipe, currently commented out until Runtime errors